medusa/process_tv.py

```diff
@@ -441,9 +441,9 @@ def unrar(self, path, rar_files, force=False):
 
                     # Skip extraction if any file in archive has previously been extracted
                     skip_extraction = False
-                    for file_in_archive in [os.path.basename(each.filename)
-                                            for each in rar_handle.infolist()
-                                            if not each.isdir()]:
+                    for file_in_archive in [os.path.basename(each)
+                                            for each in rar_handle.namelist()
+                                            if not os.path.isdir(each)]:
```

Comment:
I'd suggest to do something like this:
```
for file_in_archive in (each
                        for each in rar_handle.namelist()
                        if is_media_file(each)):
```
....
```
if not force and self.already_postprocessed(os.path.basename(file_in_archive)):
```
This would avoid breaking the os.path.isfile() check later on as well.